catalog/cataloger_diff.go

```diff
@@ -274,40 +310,44 @@ func lineageCommitIDByBranchID(lineage []lineageCommit, branchID int64) CommitID
 	return UncommittedID
 }
 
-func evaluateFromParentElementDiffType(childBranchID int64, childLastFromParentCommitID CommitID, childLineage []lineageCommit, parentEnt *DBScannerEntry, matchedChild *DBScannerEntry) DifferenceType {
+func evaluateFromParentElementDiffType(targetBranchID int64, targetLastSyncCommitID CommitID, targetLastSyncLineage []lineageCommit, sourceEntry *DBScannerEntry, targetEntry *DBScannerEntry) DifferenceType {
 	// both deleted - none
-	if parentEnt.IsDeleted() && (matchedChild == nil || matchedChild.IsDeleted()) {
+	if sourceEntry.IsDeleted() && (targetEntry == nil || targetEntry.IsDeleted()) {
 		return DifferenceTypeNone
 	}
 
 	// same entry - none
-	if matchedChild != nil && parentEnt.IsDeleted() == matchedChild.IsDeleted() && parentEnt.Checksum == matchedChild.Checksum {
+	if targetEntry != nil && sourceEntry.IsDeleted() == targetEntry.IsDeleted() && sourceEntry.Checksum == targetEntry.Checksum {
 		return DifferenceTypeNone
 	}
 
-	// parent not changed - none
-	commitIDByChildLineage := lineageCommitIDByBranchID(childLineage, parentEnt.BranchID)
-	parentChangedAfterChild := parentEnt.ChangedAfterCommit(commitIDByChildLineage)
-	if !parentChangedAfterChild {
-		return DifferenceTypeNone
+	// target entry not modified based on commit of the last sync lineage - none
+	commitIDByLineage := lineageCommitIDByBranchID(targetLastSyncLineage, sourceEntry.BranchID)
+	if commitIDByLineage > UncommittedID {
+		parentChangedAfterChild := sourceEntry.ChangedAfterCommit(commitIDByLineage)
+		if !parentChangedAfterChild {
+			return DifferenceTypeNone
+		}
```

Comment:
As discussed, this code will change in another PR (I don't understand it).

Reply:
thanks - reverted and will be addressed by PR #845